lib/remoteproc/remoteproc_virtio.c

```diff
@@ -30,6 +30,19 @@ static void rproc_virtio_virtqueue_notify(struct virtqueue *vq)
 	rpvdev->notify(rpvdev->priv, vring_info->notifyid);
 }
 
+static int rproc_virtio_wait_notified(struct virtio_device *vdev,
+				      struct virtqueue *vq)
+{
+	struct remoteproc_virtio *rpvdev;
+	struct virtio_vring_info *vring_info;
+	unsigned int vq_id = vq->vq_queue_index;
+
+	rpvdev = metal_container_of(vdev, struct remoteproc_virtio, vdev);
+	vring_info = &vdev->vrings_info[vq_id];
+
+	return rpvdev->wait_notified(rpvdev->priv, vring_info->notifyid);
```

Comment:
should we check wait_notified isn't NULL before invoke

Reply:
Done